lib/src/select/Select.tsx

```diff
@@ -562,14 +481,31 @@ const DxcSelect = React.forwardRef<RefType, SelectPropsType>(
                 ref={selectOptionsListRef}
                 role="listbox"
                 aria-multiselectable={multiple}
+                aria-orientation="vertical"
               >
                 {searchable && (filteredOptions.length === 0 || !filteredGroupsHaveOptions()) ? (
                   <OptionsSystemMessage>
                     <NoMatchesFoundIcon>{selectIcons.searchOff}</NoMatchesFoundIcon>
                     No matches found
                   </OptionsSystemMessage>
                 ) : (
-                  optional && !multiple && <Option option={optionalEmptyOption} index={0} />
+                  optional &&
+                  !multiple && (
+                    <Option
+                      id={`option-${0}`}
+                      option={optionalEmptyOption}
+                      onClick={handleOptionOnClick}
+                      multiple={multiple}
```

Comment:
Following the comment above, we can call `handleOptionOnClick(optionalEmptyOption)` here.